dockers/docker-dhcp-relay/wait_for_intf.sh.j2

```diff
@@ -30,9 +30,12 @@ function wait_until_iface_ready
 }
 
 
-# Wait for all interfaces to come up and have IPv4 addresses assigned
+# Wait for all interfaces to come up and have IPv4 addresses assigned. Skip
+# waiting on interfaces that are not configured with an IPv4 address.
 {% for (name, prefix) in INTERFACE %}
+{% if prefix != '0.0.0.0/0' %}
```

Comment:
Is this a special use-case where you actually set the IP to "0.0.0.0"? By default, if there is no IP configured, there should not even be an INTERFACE object, only a PORT object should be present.

Reply:
Explicitly setting the IP address to "0.0.0.0" to imply an interface has no IP address is not a clean solution.

In the next week or so, the framework for VRF will be added to SONiC. With this change, the INTERFACES table will contain interfaces without any IP address assigned. Once this change is merged, you should be able to key off of whether or not the interface has an IP address assigned here, which is a more appropriate solution. Please see the design docs [here](https://github.com/Azure/SONiC/pull/242) for reference.

Reply:
Setting the IP address of an interface to 0.0.0.0 is valid config and all linux cli deals with it and accepts it. Application can also listen on 0.0.0.0 and 0.0.0.0 is also used as source IP (eg DHCP).

Not sure why you think it shouldn't be allowed or why it should be restricted on sonic side.